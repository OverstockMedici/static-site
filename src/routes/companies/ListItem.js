import React from 'react';

const ListItem = ({ link, name, image, color, description }) => {
    return (
        <a
            data-testid="list"
            href={link}
            target="_blank"
            rel="noopener noreferrer"
        >
            <img src={image} alt={`${name} logo`} />
            <div
                data-testid="list-style"
                className="company-description"
                style={{ borderColor: `${color}` }}
            >
                {description}
            </div>
        </a>
    );
};

export default ListItem;
