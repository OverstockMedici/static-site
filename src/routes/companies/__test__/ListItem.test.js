import React from 'react';
import ReactDOM from 'react-dom';
import ListItem from './../ListItem';

import { render, cleanup } from '@testing-library/react';
import { getByAltText } from '@testing-library/dom';

import renderer from 'react-test-renderer';

afterEach(cleanup);

const listItemProps = {
    link: 'https://stripe.com/',
    name: 'Stripe',
    image: 'https://stripe.com/img/v3/home/social.png/',
    color: '#5160ff',
    description:
        'Stripe makes it easy to perform online financial transactions, processing billions of dollars a year for thousands of businesses',
};
const { link, name, image, color, description } = listItemProps;

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ListItem />, div);
});

it('renders ListItem correctly', () => {
    const { getByTestId, container } = render(
        <ListItem {...{ link, name, image, color, description }} />,
    );

    const imgMetaData = getByAltText(container, 'Stripe logo');
    const listComponentMetaData = getByTestId('list-style');

    expect(imgMetaData.alt).toBe('Stripe logo');
    expect(listComponentMetaData.style.borderColor).toBe(color);
    expect(getByTestId('list')).toHaveTextContent(description);
});

it('matches snapshot', () => {
    const listItemTree = renderer
        .create(<ListItem {...{ link, name, image, color, description }} />)
        .toJSON();

    expect(listItemTree).toMatchSnapshot();
});
