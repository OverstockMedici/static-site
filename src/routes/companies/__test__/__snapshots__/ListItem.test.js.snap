// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<a
  data-testid="list"
  href="https://stripe.com/"
  rel="noopener noreferrer"
  target="_blank"
>
  <img
    alt="Stripe logo"
    src="https://stripe.com/img/v3/home/social.png/"
  />
  <div
    className="company-description"
    data-testid="list-style"
    style={
      Object {
        "borderColor": "#5160ff",
      }
    }
  >
    Stripe makes it easy to perform online financial transactions, processing billions of dollars a year for thousands of businesses
  </div>
</a>
`;
